app/models/spot.rb

```diff
@@ -21,17 +21,44 @@
 class Spot < ApplicationRecord
   MAX_SPOTS_COUNT = ENV['SPOTS_COUNT_LIMIT'].to_i
 
+  attr_reader :matches
+
   belongs_to :user
   belongs_to :topic
 
   validates :title, :latitude, :longitude, :radius, presence: true
   validates :radius, numericality: { greater_than: 0 }
   validate  :spots_limit
 
+  after_create :save_compatible_matches
+
+  acts_as_mappable lat_column_name: :latitude,
+                   lng_column_name: :longitude
+
   private
 
   def spots_limit
     return unless user.spots.count >= MAX_SPOTS_COUNT
     errors.add(:base, I18n.t('api.errors.spot_limit'))
   end
+
+  def save_compatible_matches
```

Comment:
Maybe you can get rid of `save_matches` method and after a little of refactor this method could look like this:
```
def create_matches
    compatible_spots.each do |spot|
        spot.matches.create!(first_user_id: user_id, second_user_id: spot.user_id)
    end
end
```